compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -175,15 +228,57 @@ struct GPUSetEncodingOpLoweringConversion
       assert(packedShape == targetShape);
     }
 
-    // TODO(lialan): create expand_shape. Take LHS as an example:
-    // 16x4xf32 -> 16x1x4x1. Because the vector size used in the intrinsic is
-    // 1x1.
-    // For C-Matrix (i.e., ACC), it is 16x16xf32 -> 4x4x16x1xf32. Because the
-    // vector size is 4x1.
+    // Check that the dimensions of the matrix can be divided by the tile shape,
+    // if not then bail out.
+    auto sourceShape = packOp->getResult().getType().getShape();
+    assert(sourceShape.size() == 4);
+    // inner most two dimensions must be divisible.
+    if (sourceShape[2] % innerTiles[0] != 0 ||
+        sourceShape[3] % innerTiles[1] != 0) {
+      return failure();
+    }
+
+    // Create expand_shape, to tile the inner most two dimensions.
+    llvm::SmallVector<int64_t> expandShapeShape;
+    assert(intrinsicVectorShape.size() == 2); // TODO: relax this
+    auto iT1 = intrinsicVectorShape[0];
+    auto iT2 = intrinsicVectorShape[1];
+    auto oT1 = sourceShape[2] / iT1;
+    auto oT2 = sourceShape[3] / iT2;
+    expandShapeShape = {sourceShape[0], sourceShape[1], oT1, iT1, oT2, iT2};
+    assert(expandShapeShape.size() == 6);
+    RankedTensorType expandShapeType = RankedTensorType::get(
+        expandShapeShape, encodingOp.getSourceType().getElementType());
 
-    // TODO(lialan): create linalg.transpose op.
-    // LHS: 16x1x4x1 -> 4x16x1x1 (perm = [2, 0, 3, 1])
-    // ACC: 4x4x16x1 -> 4x16x4x1 (perm = [0, 2, 1, 3])
+    SmallVector<ReassociationExprs> reassociationMap(4);
+    reassociationMap[0].push_back(rewriter.getAffineDimExpr(0));
+    reassociationMap[1].push_back(rewriter.getAffineDimExpr(1));
+    reassociationMap[2].push_back(rewriter.getAffineDimExpr(2));
+    reassociationMap[2].push_back(rewriter.getAffineDimExpr(3));
+    reassociationMap[3].push_back(rewriter.getAffineDimExpr(4));
+    reassociationMap[3].push_back(rewriter.getAffineDimExpr(5));
+    auto expandShapeOp = rewriter.create<tensor::ExpandShapeOp>(
+        loc, expandShapeType, packOp->getResult(), reassociationMap);
```

Comment:
nit: we can simplify the code using `SmallVector<ReassociationIndices>`, like what you've done in l.294.